distribution/sdk/sample-cometd/cometd-java-example/src/main/java/org/codice/ddf/sdk/cometd/AsyncClient.java

```diff
@@ -0,0 +1,231 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+package org.codice.ddf.sdk.cometd;
+
+import java.net.MalformedURLException;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+import org.cometd.bayeux.Message;
+import org.cometd.bayeux.client.ClientSessionChannel;
+import org.cometd.client.BayeuxClient;
+import org.cometd.client.transport.ClientTransport;
+import org.cometd.client.transport.LongPollingTransport;
+import org.eclipse.jetty.client.HttpClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AsyncClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AsyncClient.class);
+
+    private static final String ACTIVITIES_CHANNEL = "/ddf/activities";
+    private static final String ALL_ACTIVITIES = ACTIVITIES_CHANNEL + "/**";
+    private static final String NOTIFICATIONS_CHANNEL = "/ddf/notifications";
+    private static final String ALL_NOTIFICATIONS = NOTIFICATIONS_CHANNEL + "/**";
+    private static final String DOWNLOADS_CHANNEL = NOTIFICATIONS_CHANNEL + "/downloads";
+    private static final String QUERY_SERVICE = "/service/query";
+    private static final String COMETD_CONTEXT = "/cometd";
+    private static final String DOWNLOAD_CONTEXT = "/services/catalog/sources/";
+    private static final String DOWNLOAD_TRANSFORM = "?transform=resource";
+
+    private static String asyncClientId;
```

Comment:
Since this is only sample code it may not matter too much but making those static would introduce potential threading problems. If it's not too difficult, can we make them non-static?